app/policies/diagnosis_policy.rb

```diff
@@ -0,0 +1,21 @@
+class DiagnosisPolicy < ApplicationPolicy
+  def show?
+    if user.is_a?(User)
```

Comment:
```suggestion
    if user.present?
```

Reply:
ici le code change si c'est un User on fait `@record.in?(user&.received_diagnoses)` et si c'est un expert on fait `@record.in?(expert&.received_diagnoses)` c'est pour ça que j'avais écrit ça comme ça